src/internal/utils.ts

```diff
@@ -318,7 +318,11 @@ export function getName<U extends AnyParamConstructor<any>>(cl: U) {
   const customName = options.options?.customName;
 
   if (typeof customName === 'function') {
-    return customName(options);
+    const name: any = customName(options);
+
+    assertion(typeof name === 'string' && name.length > 0, new TypeError(`The return type of "customName" must be a string and must not be empty! ("${baseName}") [E020]`));
```

Comment:
if adding an error code, there should also be an entry in `error-warnings-details.md`

(also, code `020` is [already used](https://typegoose.github.io/typegoose/docs/guides/error-warnings-details#propoptionsdiscriminators-dosnt-support-arrays-with-more-or-less-than-1-dimension-e020))

Reply:
Oh boy. I figured there would be documentation of the error codes. Why else would you have them? 😁 And I must have looked into the docs 100's of times, I just never noticed that section. 😊 

Hope the next changes are all good. 

Scott